### ISSUE TITLE
Enhancement: [prefer-nullish-coalescing] change default value of ignoreTernaryTests to false for next major release

### ISSUE THREAD

**Issue by jguddas** [typescript-eslint/typescript-eslint#5693]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing/

### Description

This option is turned on by default due to otherwise resulting in a breaking change.

https://github.com/typescript-eslint/typescript-eslint/pull/4965#issuecomment-1164457748 - @JoshuaKGoldberg  

### Fail

```typescript
.
```

### Pass

```typescript
.
```

### Additional Info

Places where `true` would need to be replaced with `false`.
https://github.com/typescript-eslint/typescript-eslint/blob/63cba5f4c1884e102927b3b14b18a00e96ac63a1/packages/eslint-plugin/docs/rules/prefer-nullish-coalescing.md?plain=1#L61
https://github.com/typescript-eslint/typescript-eslint/blob/09c04fb0d5a5dee0dc266abb2e21aa9a1c489712/packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts#L62

**Comment by Josh-Cena** [comment]
Fixed in #7110
